### PR TITLE
[15.0][FIX] product_assortment: Product black_list_domain invalidates whitelist_product_ids. New cache methods to performance

### PR DESCRIPTION
**Pull request by carlosdauden**
Fw-port https://github.com/OCA/product-attribute/pull/1272

@Tecnativa
ping @sergio-teruel @CarlosRoca13

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]

**Review by sergio-teruel — APPROVED** [review]

File: product_assortment/models/res_partner.py [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+# Copyright 2021 Tecnativa - Carlos Dauden
 # License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
-from odoo import models
+from odoo import api, fields, models
 
 
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
+    applied_assortments_ids = fields.Many2many(
```
Comment (pedrobaeza) [inline product_assortment/models/res_partner.py]: Please no plurals here for avoiding constant typos:

```suggestion
    applied_assortment_ids = fields.Many2many(
```

File: product_assortment/data/ir_cron.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    >
+        <field name="name">Product assortment recompute all partbers</field>
+        <field name="model_id" ref="model_ir_filters" />
+        <field name="binding_model_id" ref="model_ir_filters" />
+        <field name="state">code</field>
+        <field name="code">model.search([])._compute_all_partner_ids()</field>
+    </record>
+
+    <!--  Cron to recompute all partners. (Not active by default).  -->
+    <record id="ir_cron_product_assortment_recompute_all_partners" model="ir.cron">
```
Comment (pedrobaeza) [inline product_assortment/data/ir_cron.xml:17]: Is the cron really needed (and the server action)?

Comment (carlosdauden) [reply product_assortment/data/ir_cron.xml:17]: Force to recompute all partners to take into account changes in related partner fields models.
P.E.: Domain includes tag_ids.name contains "**" and user changes tag name from tags view

Comment (pedrobaeza) [reply product_assortment/data/ir_cron.xml:17]: OK

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge major

**Pull request merged**